streams/src/main/java/org/apache/kafka/streams/kstream/StateStoresSupplier.java

```diff
@@ -0,0 +1,12 @@
+package org.apache.kafka.streams.kstream;
+
+import org.apache.kafka.streams.state.StoreBuilder;
+
+import java.util.Collection;
+import java.util.Collections;
+
+public interface StateStoresSupplier {
```

Comment:
This is a meta comment: 

Let TransfomerSupplier / ProcessorSupplier to extend this class sounds a bit weird to me, since they are still just indicating "connecting to these stores" rather than "supplying these stores" and users still need to create the StoreBuilder object. So at least I'd suggest we rename this to `StateStoreConnector` to distinguish with the `StoreSupplier` which is used in Materialized for users to plug-in a customizable storage engine, and "StoreBuilder" itself as well.

Reply:
I completely agree.  I struggled with naming this, and I think `StateStoreConnector` is a much better name, especially given how the word "connect" is used to refer to this same concept in other parts of the code.